feanil/remove legacy learner sequence pages

feanil opened the pull request:
This PR should complete the legacy sequences deprecation: https://github.com/openedx/edx-platform/issues/35803

It drops all the old html templates for rendering sequences and related rendering logic.  It does not drop rendering some chromeless versions of the sequences pages which are still being used by LTI.

feanil committed:
fix: Drop _get_legacy_courseware_url and related usage.

We were running some tests using this function but it is not actually
used in the running application anymore so drop those tests and remove
the function in preparation for removing the legacy courseware itself.

feanil committed:
test: Drop a legacy test.

This test tests whether or not we can load the legacy courseware page if
we have not met prerequisites.  We don't need this anymore because we
are in the process of removing those pages and the default is now to
load the MFE instead.

The underlying checks still happens as a part of the
`_has_access_course` function which calls
`lms/djangoapps/courseware/access.py:_can_view_courseware_with_prerequisites`

feanil committed:
test: Drop testing for legacy courseware UI

This has all been replaced by the learning MFE and will be removed from
the platform in subsequent commits.

For masquerade testing, the page no longer renders content and so
shouldn't be a part of this test.  The render_xblock url is what is used
by the MFE so we're still testing that the course-wide content is being
loaded correctly for content served by the learning MFE.

feanil committed:
style: Fix various linting issues.

feanil committed:
feat: Make CoursewareIndex just a redirect.

We don't need to load the old UI and so don't need all the logic related
to it, just the logic that is expected to occur around other backend
functionality like masquerading.

feanil committed:
feat: Drop unused render_accordion function and template.

This was referenced by the legacy courseware page so we don't need it
anymore.

ormsbee commented on lms/djangoapps/courseware/views/index.py:
```diff
@@ -87,475 +49,70 @@ def enable_unenrolled_access(self):
+        # Course load to resolve chapters/sections
+        with modulestore().bulk_operations(self.course_key):
+            course = get_course_with_access(
+                request.user, 'load', self.course_key,
```
Nit: I realize this is copy-pasted from elsewhere, but please put these params on separate lines.

ormsbee commented on lms/djangoapps/courseware/views/index.py:
```diff
@@ -87,475 +49,70 @@ def enable_unenrolled_access(self):
+        with modulestore().bulk_operations(self.course_key):
+            course = get_course_with_access(
+                request.user, 'load', self.course_key,
+                depth=CONTENT_DEPTH,
```
Nit: FWIW, I think this is the only place it's being used now, so just putting `2` here and getting rid of the constant is probably the clearest thing.

ormsbee commented on lms/djangoapps/courseware/views/index.py:
```diff
@@ -87,475 +49,70 @@ def enable_unenrolled_access(self):
+        # Get the chapter, section and unit blocks so that we can redirect to the right content
+        # location in the MFE
+        section_location = None
+        if chapter and section:
```
We should either update the param names here to match modern terminology (i.e. chapter -> section, section -> subsection), or leave a comment explaining the old terminology and why "section" doesn't mean what anyone would think it means here.

ormsbee commented on lms/djangoapps/courseware/tests/test_course_survey.py:
```diff
@@ -94,14 +95,32 @@ def _assert_no_redirect(self, course):
+        )
+        assert resp.status_code == 200
+
+        # Make sure we are redirected to the MFE
```
```suggestion
        # Make sure we are redirected to the MFE for courseware
```

ormsbee commented on lms/djangoapps/courseware/tests/test_course_survey.py:
```diff
@@ -94,14 +95,32 @@ def _assert_no_redirect(self, course):
+        resp = self.client.get(
+            reverse(
+                'courseware',
+                kwargs={'course_id': str(course.id)}
```
Should this also test passing in params for chapter and section?

feanil replied:
This was only testing `reverse('courseware')` previously so I just updated the test. I can add further testing here but that would be increasing coverage not an update of the tests to match the new default functionality.  Let me know if want me to do that.

ormsbee replied:
I mostly just want to make sure that the new redirect-to-mfe-based-on-params code in `CoursewareIndex.get` is tested somewhere. Probably in `test_views.py`?

feanil replied:
I added a test for each of the URL variations, `courseware`, `courseware_section`(previously `courseware_chapter`) and `courseware_subsection` (previousl `courseware_section`).  The masquerade logic is being tested in the masquerade app.

ormsbee commented on lms/djangoapps/courseware/tests/test_course_survey.py:
```diff
@@ -94,14 +95,32 @@ def _assert_no_redirect(self, course):
+            )
+        )
+        assert resp.status_code == 302
+        expected_redirect_url = make_learning_mfe_courseware_url(
```
Should this be in a separate test? Since the name of this method is `_assert_no_redirect()`?

Though I'm a little confused about how this is working anyhow, and where these methods are being called from...?

feanil replied:
That name was a bit misleading, I've updated it to match what that helper is actually doing.

ormsbee reviewed (changes requested):
Just a few minor things. It's probably worth advertising this in #cc-risky-changes now (or pinging it again if you advertised it a while ago).

feanil committed:
fix: Update naming for courseware section/subsection.

These used to be named chapter and section before but we want to update
the courseware index view to use the new names if it's gonna stick
around.
